cwms-http-client/src/main/java/mil/army/usace/hec/cwms/http/client/HttpRequestBuilderImpl.java

```diff
@@ -217,8 +220,26 @@ public HttpRequestExecutor withMediaType(String type) {
         }
     }
 
+    public final String download(String urlString) throws IOException {
+        StringBuilder result = new StringBuilder();
+
+        URL url = new URL(urlString);
+        HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+        connection.setRequestMethod("GET");
+
+        try(BufferedReader reader = new BufferedReader(new InputStreamReader(connection.getInputStream()))) {
+            String line;
+            while((line = reader.readLine()) != null) {
+                result.append(line);
+            }
+        }
```

Comment:
This part could be replaced with some sort of static readFully call.  I'm not sure what libraries are available.  If there isn't already some sort of readFully you should refactor one

Reply:
Now using OkHttpClient to build and execute request. Should there be timeout logic in place for connection/read. I know the purpose of this feature was to circumvent the timeout but should there not be any timeout at all?

Reply:
Might consider using read(char[]) instead of readLine like:   
```
  private static String readFully2(InputStream inputStream) throws IOException {
        try (BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(inputStream))) {
            StringBuilder stringBuilder = new StringBuilder();
            char[] buffer = new char[1024];
            int read;
            while ((read = bufferedReader.read(buffer)) != -1) {
                stringBuilder.append(buffer, 0, read);
            }
            return stringBuilder.toString();
        }
        
    }
```